door-runtime/src/jsMain/kotlin/com/ustadmobile/door/DatabaseChangeMonitor.kt

```diff
@@ -0,0 +1,36 @@
+package com.ustadmobile.door
+
+import kotlinx.coroutines.GlobalScope
+import kotlinx.coroutines.Job
+import kotlinx.coroutines.async
+import kotlinx.coroutines.delay
+import wrappers.SQLiteDatasourceJs
+
+/**
+ * This class is used to listen for all database changes and persist database to the indexedDB
+ * after some delays.
+ */
+class DatabaseChangeMonitor(database: DoorDatabase, private val datasource: SQLiteDatasourceJs,
```

Comment:
I think the name 'DatabaseChangeMonitor' doesn't really match it's purpose. I think we could call it SaveToIndexedDbChangeListener